compiler/src/iree/compiler/Dialect/Flow/IR/FlowOps.cpp

```diff
@@ -645,16 +645,17 @@ LogicalResult DispatchTensorLoadOp::reifyResultShapes(
         continue;
       }
       shape.push_back(
-          getValueOrCreateConstantIndexOp(b, getLoc(), mixedSize.value()));
+          (Value)getValueOrCreateConstantIndexOp(b, getLoc(),
+                                                 mixedSize.value()));
     }
   } else {
     // Result size matches the source size (no slicing).
     unsigned dynamicIdx = 0;
     for (int64_t dim : getType().getShape()) {
       if (dim == ShapedType::kDynamic) {
-        shape.push_back(getSourceDims()[dynamicIdx++]);
+        shape.push_back((Value)getSourceDims()[dynamicIdx++]);
       } else {
-        shape.push_back(b.create<arith::ConstantIndexOp>(getLoc(), dim));
+        shape.push_back((Value)b.create<arith::ConstantIndexOp>(getLoc(), dim));
```

Comment:
This must return an OpFoldResult.

From the documentation of `reifyResultShapes`:
```
        `reifiedReturnShapes` is populated with one vector per op result. Each
        of those vectors contains an OpFoldResult for each dimension of the
        shaped type. In case a dimension in the type is static, the
        corresponding entry is an IntegerAttr. Otherwise, it is a Value. The
        given builder may be used to insert ops that compute result shapes.
```

In this particular case here:
```
        shape.push_back(b.getIndexAttr(dim));
```

(same for a few more places below)

Reply:
So

```
        shape.push_back((Value)b.create<arith::ConstantIndexOp>(getLoc(), dim));
```

should be changed to

```
        shape.push_back(b.getIndexAttr(dim));
```

?